Unhelpful error in dynamic keyword argument interpolation

### Bug description

Consider:

```python
"""..."""


def test_symbol_coverage(thing):
    """..."""
    amount = 7
    dynamic_kwarg = {f"{thing}_amount": amount}
    func_taking_keyword_arguments(**dynamic_kwarg)


def func_taking_keyword_arguments(lobster_amount=0, spam_amount=0):
    """..."""
    del lobster_amount
    del spam_amount
```

Running `pylint bug.py` yields

```text
************* Module bug
bug.py:8:4: E1123: Unexpected keyword argument 'Uninferable_amount' in function call (unexpected-keyword-arg)

------------------------------------------------------------------
Your code has been rated at 2.86/10 (previous run: 2.86/10, +0.00)
```

Pylint appears to be failing to infer the value of `thing`, but then interpolating it into the format string, serialising the uninferable thing `thing` as the literal string `Uninferable`. This is then taken with the other text in the format string literal and matched against the parameters of the given function, which fails.

I think pylint should do nothing here, stopping flow analysis as soon as it encounters an uninferable variable.

### Configuration

```ini

```

### Command used

```shell
pylint bug.py
```

### Pylint output

```python
************* Module bug
bug.py:8:4: E1123: Unexpected keyword argument 'Uninferable_amount' in function call (unexpected-keyword-arg)

------------------------------------------------------------------
Your code has been rated at 2.86/10 (previous run: 2.86/10, +0.00)
```

### Expected behavior

No error.

### Pylint version

```shell
pylint 4.0.4
astroid 4.0.3
Python 3.14.2 (main, Dec  5 2025, 16:49:16) [Clang 17.0.0 (clang-1700.4.4.1)]
```

### OS / Environment

macOS Tahoe 26

### Additional dependencies

```python

```

--- a/ChangeLog
+++ b/ChangeLog
@@ -7,6 +7,13 @@ What's New in astroid 4.1.0?
 ============================
 Release date: TBA
 
+* Ensure ``ast.JoinedStr`` nodes are ``Uninferable`` when the ``ast.FormattedValue`` is
+  ``Uninferable``. This prevents ``unexpected-keyword-arg`` messages in Pylint
+  where the ``Uninferable`` string appeared in function arguments that were
+  constructed dynamically.
+
+  Closes pylint-dev/pylint#10822
+
 * Add support for type constraints (`isinstance(x, y)`) in inference.
 
   Closes pylint-dev/pylint#1162

--- a/astroid/nodes/node_classes.py
+++ b/astroid/nodes/node_classes.py
@@ -4740,6 +4740,9 @@ class FormattedValue(NodeNG):
                     uninferable_already_generated = True
                 continue
             for value in self.value.infer(context, **kwargs):
+                if value is util.Uninferable:
+                    yield util.Uninferable
+                    return
                 value_to_format = value
                 if isinstance(value, Const):
                     value_to_format = value.value

--- a/tests/test_inference.py
+++ b/tests/test_inference.py
@@ -7154,3 +7154,21 @@ def test_joined_str_returns_string(source, expected) -> None:
         inferred[0].value.startswith(expected)
     else:
         assert inferred[0] is Uninferable
+
+
+def test_joined_str_uninferable() -> None:
+    """`thing` is not known, therefore the joinedstring should be
+    Uninferable
+    """
+    code = """
+        def hey(thing):
+            return thing
+
+        f"Hello {hey()}"
+    """
+    joined_str = extract_node(code)
+    constant_value, formatted_value = joined_str.values
+    assert constant_value.value == "Hello "
+    assert formatted_value.value.as_string() == "hey()"
+    inferred = next(joined_str.infer())
+    assert inferred is util.Uninferable

